diff --git gcc-9.3.0-orig/gcc/config/phoenix.h gcc-9.3.0/gcc/config/phoenix.h
index c1aad95..1985dde 100644
--- gcc-9.3.0-orig/gcc/config/phoenix.h
+++ gcc-9.3.0/gcc/config/phoenix.h
@@ -31,3 +31,5 @@ along with GCC; see the file COPYING3.  If not see
 /* This will prevent selecting 'unsigned long int' instead of 'unsigned int' as 'uint32_t' in stdint-newlib.h. */
 #undef STDINT_LONG32
 #define STDINT_LONG32		0
+/* libitm is not supported on Phoenix - disable it in crtstuff */
+#define USE_TM_CLONE_REGISTRY 0
